aten/src/ATen/mps/MPSAllocator.mm

```diff
@@ -39,6 +39,10 @@
   TORCH_CHECK(ratio >= 0.0 && ratio <= default_high_watermark_upper_bound, "invalid high watermark ratio ", ratio);
   m_max_total_allowed_size = (ratio == 0.0) ? std::numeric_limits<size_t>::max() :
                              static_cast<size_t>(ratio * (double)max_device_size());
```

Comment:
Is there value to have this always printed as error instead of when debugging ? Maybe helpful in long running networks if they eventually hit ?

Reply:
It's only printed when the env var `PYTORCH_DEBUG_MPS_ALLOCATOR=1` is enabled. 
The high/low watermark values are printed for debugging purpose only when a python-level API like `torch.mps.set_per_process_memory_fraction()` is called.
Also, we redirect our prints to cerr instead of cout as instructed in other reviews by Meta developers.
All these debug prints will be replaced by `memory_stats` feature at some point:
https://pytorch.org/docs/stable/generated/torch.cuda.memory_stats.html